### PR TITLE
Add pn.panel and ensure all Panes have layout attribute

### PR DESCRIPTION
**Pull request by philippjfr**
@jbednar Please test this extensively.

- [x] Implements https://github.com/pyviz/panel/issues/124
- [x] Updates unit tests
- [x] Update docs

### CODE REVIEW

File: panel/layout.py [hunk 1]
```diff
@@ -152,23 +158,23 @@ def pop(self, index):
         self.objects = new_objects
```
Comment (jbednar) [inline panel/layout.py]: Should we add Layout = Panel for now?  Or is it unlikely that user code would have used Layout by that name?

File: panel/layout.py [hunk 2]
```diff
@@ -313,3 +320,6 @@ def _get_model(self, doc, root=None, parent=None, comm=None):
         model = self._bokeh_model(**self._init_properties())
         self._link_params(model, ['width', 'height'], doc, root, comm)
         return model
+
+
+PaneBase._default_layout = Row
```
Comment (jbednar) [inline panel/layout.py]: Should this not be a public class Parameter?  Seems like something one could legitimately want to customize.

File: panel/pane.py [hunk 3]
```diff
@@ -24,6 +24,31 @@
...
+
+    Returns
+    -------
+    layout: Viewable
+       A Viewable representation of the input object
+    """
+    if isinstance(obj, PaneBase):
+        return obj
+    elif isinstance(obj, Viewable):
+        return obj
```
Comment (jbednar) [inline panel/pane.py]: Isn't PaneBase a subtype of Viewable?  So why test PaneBase explicitly?

File: panel/layout.py [hunk 4]
```diff
@@ -129,19 +129,25 @@ def _get_model(self, doc, root=None, parent=None, comm=None):
     def __getitem__(self, index):
         return self.objects[index]
 
+    def __len__(self):
```
Comment (jbednar) [inline panel/layout.py]: Shouldn't `pane = Pane(pane, _temporary=True)` in _get_objects above be `pane = create(pane, _temporary=True)`?

File: panel/layout.py [hunk 5]
```diff
@@ -10,14 +10,14 @@
                            WidgetBox as BkWidgetBox, Spacer as BkSpacer)
 from bokeh.models.widgets import Tabs as BkTabs, Panel as BkPanel
 
-from .pane import Pane, PaneBase
+from .pane import create, Pane, PaneBase
```
Comment (jbednar) [inline panel/layout.py]: Can this import of Pane be deleted?  If not, seems like it should be marked `# noqa`.

File: panel/pane.py (hunk 3 above)
Comment (philippjfr) [reply panel/pane.py]: No reason now, just didn't tidy it up.

File: panel/layout.py (hunk 1 above)
Comment (philippjfr) [reply panel/layout.py]: Seems unlikely that any user would have used an abstract baseclass.

**Comment by philippjfr** [comment]
Always prefer if you warn me about making edits to a branch I'm actively working on. You can now also leave suggested edits directly in the GitHub, which let's me apply the changes when I'm ready.

File: panel/layout.py (hunk 4 above)
Comment (philippjfr) [reply panel/layout.py]: Actually no, this was deliberate.

**Comment by jbednar** [comment]
@jlstevens, consider code like this:

```
header = pn.Row(pn.create(oggm_logo, width=170), pn.Spacer(width=50), 
                pn.Column(pn.Pane(title, height=25, width=400), pn.Spacer(height=-15), pn.create(instruction, width=500)),
                pn.Spacer(width=180), pn.Column(pn.create(dyn_count), clear_button, pn.Spacer(height=-15)), 
                pn.create(pv_logo, width=80))
```

To me `create` reads oddly there; I'd rather see `panel` so that I can see that it's a row of rows, spacers, and panels, not a row of creates...

File: panel/layout.py (hunk 4 above)
Comment (philippjfr) [reply panel/layout.py]: Nevermind, it wasn't.

File: examples/user_guide/Panes.ipynb [hunk 6]
```diff
@@ -61,14 +78,23 @@
...
    "source": [
-    "pane.object = '<b>Some updated text</b>'"
+    "pane.object = 'Some updated text'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "One thing to note is that since a ``Pane`` may define multiple views, when working with an explicitly constructed ``Pane`` care should be taken to compose the component of the ``Pane.layout`` manually."
```
Comment (jbednar) [inline examples/user_guide/Panes.ipynb]: the components?

**Pull request merged**